### PR TITLE
Xrt m2 upgrade

### PR DESCRIPTION
**Pull request by nrwslac**
Controls software to support the removal of ELMO hardware and installation of Beckhoff hardware. Modbus no longer active.

### CODE REVIEW

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU [hunk 1]
```diff
@@ -480,6 +630,89 @@ mcReadParameterPitchMR1L3(Axis:=M5.Axis,
 
 nEncRefPitchMR1L3 := LREAL_TO_UDINT(ABS(fEncRefPitchMR1L3_urad) * fEncLeverArm_mm);
 
+//Beckhoff MR1L4
+MR1L4.fbRunHOMS(stYup:=M7,
```
Comment (klauer) [inline HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:634]: Reminder to reviewers: FB_RunHOMS comes from lcls-twincat-optics (https://pcdshub.github.io/lcls-twincat-optics/v0.2.1/lcls-twincat-optics_lcls_twincat_optics_plc_source.html#fb-runhoms)

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU [hunk 2]
```diff
@@ -177,6 +177,176 @@ VAR
...
+	{attribute 'TcLinkTo' := '.fbRunHOMS.bSTOEnable1:=TIIB[EL1004_M1L4_STO]^Channel 1^Input;
+                              .fbRunHOMS.bSTOEnable2:=TIIB[EL1004_M1L4_STO]^Channel 2^Input;
+	                          .fbRunHOMS.stYupEnc.Count:=TIIB[EL5042_M1L4_Yupdwn]^FB Inputs Channel 1^Position;
+	                          .fbRunHOMS.stYdwnEnc.Count:=TIIB[EL5042_M1L4_Yupdwn]^FB Inputs Channel 2^Position;
+	                          .fbRunHOMS.stXupEnc.Count:=TIIB[EL5042_M1L4_Xupdwn]^FB Inputs Channel 1^Position;
+	                          .fbRunHOMS.stXdwnEnc.Count:=TIIB[EL5042_M1L4_Xupdwn]^FB Inputs Channel 2^Position'}
+	{attribute 'pytmc' := '
+		pv: MR1L4:HOMS
+	'}
+	MR1L4 : DUT_HOMS;
```
Comment (klauer) [inline HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:245]: Reminder to reviewers: DUT_HOMS comes from lcls-twincat-optics: https://pcdshub.github.io/lcls-twincat-optics/v0.2.1/lcls-twincat-optics_lcls_twincat_optics_plc_source.html#dut-homs

**Review by klauer — COMMENTED** [review]
Partial review, things mostly looking good so far. Needs a bit more cleanup on removing old, unused code.

**Comment by ZLLentz** [comment]
After this is all done, it's probably finally time to rename the repo/project to match the others. Doesn't have to be during this PR.

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU [hunk 3]
```diff
@@ -599,37 +832,37 @@ END_IF
...
-			 IN_RANGE:=M2_INRANGE,
-			 OUT_RANGE:=M2_OUTRANGE);
+//fbMPS_States(GANTRY:=HOMS2_XGantry,
+//			 OUT_LS:=NOT HOMS2_XGantry.PAxis.xHiLS OR NOT HOMS2_XGantry.SAxis.xLoLS,
+//			 IN_RANGE:=M2_INRANGE,
+//			 OUT_RANGE:=M2_OUTRANGE);
 			 
-bXRT_M2_IN  := fbMPS_States.IN;
-bXRT_M2_OUT := fbMPS_States.OUT;
+bXRT_M2_IN  := Main.MR1L4.fbRunHOMS.stXupEnc.Count < 25490840;
```
[the remark this answers is not in the document]
Comment (nrwslac) [reply HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:857]: I'm not sure why magic values are ever okay.

[the remark this answers is not in the document]
Comment (klauer) [reply HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:857]: They aren't, you are 100% correct! 😁

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU [hunk 4]
```diff
@@ -572,17 +805,17 @@ nEncRefPitchMR2L3 := LREAL_TO_UDINT(ABS(fEncRefPitchMR2L3_urad) * fEncLeverArm_m
...
 
 
 //InitParams
 ////////////////////
-rtInitParams(CLK:=gFirstPass);
-IF rtInitParams.Q THEN
-	Init_XRTM2();
-	gFirstPass := FALSE;
-END_IF
+//rtInitParams(CLK:=gFirstPass);
```
Comment (klauer) [inline HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:814]: All of these commented and no-longer-valid/relevant blocks need to go away; they make this more difficult to review or just read.

File: HOMS_XRT/HOMS_XRT_PLC/POUs/PRG_CoatingProtection.TcPOU [hunk 5]
```diff
@@ -75,8 +75,8 @@ ELSE // Anything else in the middle of the range
 END_IF
 
 fbM2(
-	bMirrorIn := (HOMS2_XGantry.PAxis.diEncCnt < 25490840), 
-    rCurrentEncoderPosition := HOMS2_YGantry.PAxis.diEncCnt,
+	bMirrorIn := (Main.MR1L4.fbRunHOMS.stXupEnc.Count < 25490840), 
+    rCurrentEncoderPosition := Main.nEncCntYupMR1L4,
```
Comment (klauer) [inline HOMS_XRT/HOMS_XRT_PLC/POUs/PRG_CoatingProtection.TcPOU:79]: `Main.nEncCntYupMR1L4` is
```
nEncCntYupMR1L4 := ULINT_TO_UDINT(MR1L4.fbRunHOMS.stYupEnc.Count);
```
but you use `Main.MR1L4.fbRunHOMS.stXupEnc.Count` directly above? Any reason for not being consistent between the two?

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU [hunk 6]
```diff
@@ -177,6 +177,176 @@ VAR
...
+	'}
+	fbYRMSErrorMR1L4 : FB_RMSWatch;
+	fMaxYRMSErrorMR1L4 : LREAL;
+	fMinYRMSErrorMR1L4 : LREAL;
+
+	{attribute 'pytmc' := '
+		pv: MR1L4:HOMS:ENC:X
+	'}
+	fbXRMSErrorMR1L4 : FB_RMSWatch;
+	fMaxXRMSErrorMR1L4 : LREAL;
```
Comment (klauer) [inline HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:260]: Do these not go to EPICS? Are we just using them to look at in TwinCAT when logging in... ?

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU [hunk 7]
```diff
@@ -342,51 +512,31 @@ VAR
 	fEncRefPitchMR2L3_urad : LREAL; // Current Pitch encoder offset in urad
 	
 	//ELMO M2
-	XRTM2PitchControl	    :	FB_PitchControlOld;
-	XRTM2Y_GantryControl	:	FB_Gantry;
-	XRTM2X_GantryControl	:	FB_Gantry;
+	//XRTM2PitchControl	    :	FB_PitchControlOld;
```
Comment (klauer) [inline HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:515]: More old references to remove

**Review by klauer — CHANGES_REQUESTED** [review]
I think the core of this looks good, but I'd like to see all of the old references to old code cleaned up before merging this back in. 

It's best to do these sorts of things at refactor time rather than down the line, since it will only be more confusing then.

File: HOMS_XRT/HOMS_XRT_PLC/POUs/PRG_CoatingProtection.TcPOU (hunk 5 above)
Comment (nrwslac) [reply HOMS_XRT/HOMS_XRT_PLC/POUs/PRG_CoatingProtection.TcPOU:79]: looks equivalent, changes made.

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU (hunk 4 above)
Comment (nrwslac) [reply HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:814]: deleted more commented out code

File: HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU (hunk 6 above)
Comment (nrwslac) [reply HOMS_XRT/HOMS_XRT_PLC/POUs/MAIN.TcPOU:260]: not sure what these are used for, theres a lot of testing/lingering code in this project that my have been future plans or previous test software.

**Review by klauer — APPROVED** [review]
Looks much better - nice work @nrwslac 👍

**Pull request merged**